### PR TITLE
HeredocToNowdocFixer - Handle space in heredoc token

### PR DESCRIPTION
**Pull request by SpacePossum**
Closes https://github.com/FriendsOfPHP/PHP-CS-Fixer/issues/1869

ping @GrahamCampbell and @gharlan

### CODE REVIEW

**Comment by GrahamCampbell** [comment]
:+1:

**Comment by GrahamCampbell** [comment]
Some of these test cases are invalid php.

**Comment by GrahamCampbell** [comment]
```
There was 1 failure:
1) Symfony\CS\Tests\Fixer\Symfony\HeredocToNowdocFixerTest::testFix with data set #8 ('<?php\n$html = <<<'HTML'\na\nHTM...HTML;\n')
Failed asserting that '
Parse error: syntax error, unexpected ',' in /home/travis/build/FriendsOfPHP/PHP-CS-Fixer/cs_fixer_tmp_Khtpgj on line 6
Errors parsing /home/travis/build/FriendsOfPHP/PHP-CS-Fixer/cs_fixer_tmp_Khtpgj
' is null.
/home/travis/build/FriendsOfPHP/PHP-CS-Fixer/Symfony/CS/Tests/Fixer/AbstractFixerTestBase.php:95
/home/travis/build/FriendsOfPHP/PHP-CS-Fixer/Symfony/CS/Tests/Fixer/Symfony/HeredocToNowdocFixerTest.php:29
```

**Comment by gharlan** [comment]
I think the fixer should not remove the spaces.

**Comment by GrahamCampbell** [comment]
> I think the fixer should not remove the spaces.

Not removing the spaces results in a syntax error.

**Comment by gharlan** [comment]
Oh, ok, didn't know that.

File: Symfony/CS/Tests/Fixer/Symfony/HeredocToNowdocFixerTest.php [hunk 1]
```diff
@@ -104,6 +104,36 @@ public function provideTestFixCases()
...
+EOF
+            ),
+            array(<<<'EOF'
+<?php $a = <<<'TEST'
+Foo
+TEST;
+
+EOF
+            , <<<'EOF'
+<?php $a = <<<           "TEST"
```
Comment (gharlan) [inline Symfony/CS/Tests/Fixer/Symfony/HeredocToNowdocFixerTest.php]: hmm, this is invalid. why isn't it failing in the travis build with enabled linting?

Comment (GrahamCampbell) [inline Symfony/CS/Tests/Fixer/Symfony/HeredocToNowdocFixerTest.php]: errr, not sure

Comment (GrahamCampbell) [inline Symfony/CS/Tests/Fixer/Symfony/HeredocToNowdocFixerTest.php]: see https://github.com/FriendsOfPHP/PHP-CS-Fixer/issues/1869

Comment (SpacePossum) [inline Symfony/CS/Tests/Fixer/Symfony/HeredocToNowdocFixerTest.php]:
```php
<?php $a = <<<           "TEST"
Foo
TEST;

echo $a;die;
```

runs for me, what part is invalid?

Comment (GrahamCampbell) [inline Symfony/CS/Tests/Fixer/Symfony/HeredocToNowdocFixerTest.php]: What about with single quotes?

Comment (SpacePossum) [inline Symfony/CS/Tests/Fixer/Symfony/HeredocToNowdocFixerTest.php]: same, all ok (PHP 5.5.9-1ubuntu4.14 (cli) (built: Oct 28 2015 01:34:46)

Comment (gharlan) [inline Symfony/CS/Tests/Fixer/Symfony/HeredocToNowdocFixerTest.php]: @SpacePossum is right, both are valid.

**Comment by gharlan** [comment]
> Not removing the spaces results in a syntax error.

it is valid: https://3v4l.org/GlPME

So, I still think we should not remove the spaces.

**Comment by GrahamCampbell** [comment]
> So, I still think we should not remove the spaces.

How come my other test file is getting broken then?

**Comment by gharlan** [comment]
Because the fixer does not handle the spaces correctly.

**Comment by SpacePossum** [comment]
>How come my other test file is getting broken then?

because the opening quote was misplaced;
current wrong:

```
<<< "    HTML"
```

PR fixes to:
```
<<< "HTML"
```
suggested is:
```
<<<     "HTML"
```
input
```
<<<     HTML
```

I guess?

**Comment by gharlan** [comment]
suggested is:

```php
<<<     'HTML'
```

(single quotes)

**Comment by SpacePossum** [comment]
course :)
updated, I have the feeling the replace can be done more efficient, if so maybe someone can suggest how?

**Comment by gharlan** [comment]
> I have the feeling the replace can be done more efficient

I don't think so.

👍 🚢

**Comment by GrahamCampbell** [comment]
:+1:

**Comment by GrahamCampbell** [comment]
Is this good to merge @keradus?

**Comment by keradus** [comment]
Yes, please rebase.
I got falsy impression thanks to our stalker ;)

**Comment by GrahamCampbell** [comment]
> I got falsy impression thanks to our stalker ;)

I wonder if I can get gitHub to actually delete all the mess they've made.

**Comment by keradus** [comment]
One one or another please rebase and put your thumb for #1903 ;)

**Comment by SpacePossum** [comment]
rtm :)
dunno why the test don't run atm btw

**Comment by keradus** [comment]
why based on #1896 ?

**Comment by SpacePossum** [comment]
I guess I messed up the rebase, wasn't a good weekend for development, lets see

**Comment by GrahamCampbell** [comment]
:+1:

**Comment by keradus** [comment]
Thank you @SpacePossum.

**Pull request merged**